### DISCUSSION TITLE
Old ENV VARS not being deleted, surfacing DEV keys in PRODUCTION

### DISCUSSION THREAD

**Discussion by MaximusMcCann** [vercel/community#2738] [Bug Report]
### Page to Investigate

settings/environment-variables

### Steps to Reproduce

New ENV vars delete properly, but ones I created roughly 1 month ago and have since deleted (today) are still showing up.

My issue is our **Production environment is surfacing both dev/preview keys and our production keys!**

We append `_LIVE` to all our production keys to help ensure safety, but Vercel is showing both versions of the keys still...even after fully deleting the dev/preview key.

Example key names
`NEXT_PUBLIC_SUPABASE_URL` = Dev/Preview
`NEXT_PUBLIC_SUPABASE_URL_LIVE` = Production

At one point `NEXT_PUBLIC_SUPABASE_URL` was used for all environments, but now looking to deploy we only want the `NEXT_PUBLIC_SUPABASE_URL_LIVE` one.  I have deleted `NEXT_PUBLIC_SUPABASE_URL` entirely, but the key is still present in our build :(.

Note: I have also cleared cache and Purged Data Cache.

Please help!

**Comment by markthepixel** [comment]
Same issue

**Comment by amyegan** [comment]
Hi @MaximusMcCann and @markthepixel. Can you tell me more about your deployment process? I know most people rely on the [Vercel GitHub integration](https://vercel.com/docs/concepts/deployments/git/vercel-for-github) for automatic deployments, but some use a customized CI solution or local CLI for deployments. I want to ensure I'm looking at the right thing.

If you have other details you are comfortable sharing, like specific env var names or dates when they were deleted, please let me know. Anything you can share that might help with investigation is appreciated 🙏

**Comment by MaximusMcCann** [comment]
Figured it out.  TLDR: ensure you use `.env.local` for local variables, not `.env`.

I was using a `.env` file, thus when locally running `vercel --prod -force` it would pull in the `.env` variables too, despite having .gitignored the file.
